File: luci-app-iptv-manager/luasrc/model/cbi/iptv-manager/playlist.lua
local m, s, o

m = Map("iptv", translate("Настройки плейлиста"),
    translate("Загрузка и управление IPTV плейлистом"))

m:chain("iptv")

-- Playlist name
o = m:option(Value, "playlist_name", translate("Название плейлиста"))
o.placeholder = "Мой плейлист"
o.rmempty = true

-- Playlist type
o = m:option(ListValue, "playlist_type", translate("Тип плейлиста"))
o:value("url", "По ссылке (URL)")
o:value("file", "Из файла")
o:value("provider", "Провайдер")
o:value("", "Не загружен")

-- Playlist URL
o = m:option(Value, "playlist_url", translate("URL плейлиста"))
o.placeholder = "https://example.com/playlist.m3u"
o:depends("playlist_type", "url")
o.datatype = "string"

-- Playlist source file
o = m:option(Value, "playlist_source", translate("Путь к файлу"))
o.placeholder = "/tmp/playlist.m3u"
o:depends("playlist_type", "file")

-- Provider config
o = m:option(Value, "provider_name", translate("Название провайдера"))
o:depends("playlist_type", "provider")

o = m:option(Value, "provider_login", translate("Логин провайдера"))
o:depends("playlist_type", "provider")

o = m:option(Value, "provider_pass", translate("Пароль провайдера"))
o.password = true
o:depends("playlist_type", "provider")

o = m:option(Value, "provider_server", translate("Сервер провайдера"))
o.placeholder = "http://example.com"
o:depends("playlist_type", "provider")

return m

File: luci-app-iptv-manager/htdocs/luci-static/resources/view/iptv-manager/playlist.js
'use strict';
'require view';
'require form';
'require uci';
'require ui';

return view.extend({
    load: function() {
        return Promise.all([
            uci.load('iptv')
        ]);
    },

    render: function() {
        var m, s, o;

        m = new form.Map('iptv', _('Настройки плейлиста'),
            _('Загрузка и управление IPTV плейлистом'));

        s = m.section(form.TypedSection, 'main', null, true);
        s.anonymous = true;

        o = s.option(form.Value, 'playlist_name', _('Название плейлиста'));
        o.placeholder = 'Мой плейлист';
        o.rmempty = true;

        o = s.option(form.ListValue, 'playlist_type', _('Тип плейлиста'));
        o.value('url', _('По ссылке (URL)'));
        o.value('file', _('Из файла'));
        o.value('provider', _('Провайдер'));
        o.value('', _('Не загружен'));

        o = s.option(form.Value, 'playlist_url', _('URL плейлиста'));
        o.placeholder = 'https://example.com/playlist.m3u';
        o.depends('playlist_type', 'url');

        o = s.option(form.Value, 'playlist_source', _('Путь к файлу'));
        o.placeholder = '/tmp/playlist.m3u';
        o.depends('playlist_type', 'file');

        o = s.option(form.Value, 'provider_name', _('Название провайдера'));
        o.depends('playlist_type', 'provider');

        o = s.option(form.Value, 'provider_login', _('Логин провайдера'));
        o.depends('playlist_type', 'provider');

        o = s.option(form.Value, 'provider_pass', _('Пароль провайдера'));
        o.password = true;
        o.depends('playlist_type', 'provider');

        o = s.option(form.Value, 'provider_server', _('Сервер провайдера'));
        o.placeholder = 'http://example.com';
        o.depends('playlist_type', 'provider');

        return m.render();
    }
});
